### DISCUSSION TITLE
Can someone give me an example of how they use this?

### DISCUSSION THREAD

**Discussion by crazygonuts** [Bing-su/adetailer#121] [General]
I read a bunch of people talking about how great this extension is, and then I read the readme and I just don't understand what it actually does for you. especially in the case of txt2img.  Automasking the face or hands for inpainting doesn't sound game changing to me, maybe someone can explain to me why it is?  I'm not trying to be a jerk, I hope it doesn't come off that way, but I'm just not clear on how auto detecting the mask is supposed to help me all that much.  Can someone give me some example shots of what their settings were and explain what that accomplished for you?

I'm also really unclear on what the controlnet submenu does different from the regular controlnet.

**Comment by masonintokyo** [comment]
@crazygonuts 
For example, if the face is small compared to the entire image, facial details may not be adequately rendered. This requires inpainting, which Adetailer can automate. In other cases, if you want to use lora for a face, you must always inpainting the face after image generation because lora will affect the entire image. Adetailer also automates this process.

**Comment by andupotorac** [comment]
What about this part?
"I'm also really unclear on what the controlnet submenu does different from the regular controlnet."

**Comment by andupotorac** [comment]
> I read a bunch of people talking about how great this extension is, and then I read the readme and I just don't understand what it actually does for you. especially in the case of txt2img. Automasking the face or hands for inpainting doesn't sound game changing to me, maybe someone can explain to me why it is? I'm not trying to be a jerk, I hope it doesn't come off that way, but I'm just not clear on how auto detecting the mask is supposed to help me all that much. Can someone give me some example shots of what their settings were and explain what that accomplished for you?
> 
> I'm also really unclear on what the controlnet submenu does different from the regular controlnet.

I can give you an example. So the idea is you get to run a prompt once, say with a batch of 10 images, and the need to fix faces or hands is almost entirely gone (I believe future improvements could iterate on this to fix other things, like eyes, feet, animal faces, etc).

So, let me give you a concrete example with what the prompt generated, and then what the extension did. See the hand and the face. 

![image](https://github.com/Bing-su/adetailer/assets/3269738/e682bbd5-9762-49c8-806c-6187d5da08f0)

I was able to give it a prompt with a Textual Inversion - which was used in the main prompt anyway - and fix the face. So the ADetailer prompt for the face only has this part: `(Marc2500:1), smiling, excited, hazel eyes, detailed eyes`

And for hands: `beautiful hands`


Result:
![image](https://github.com/Bing-su/adetailer/assets/3269738/5ef4b8d2-14ee-44b2-8be4-ec1ca9ce48f6)

**Comment by andupotorac** [comment]
Here's another example (ignore the fact my SD has 3 hands): 
<img width="870" alt="image" src="https://github.com/Bing-su/adetailer/assets/3269738/de575056-6213-4b10-b681-87ac40336296">
